mapie/calibration.py

```diff
@@ -0,0 +1,548 @@
+
+        Parameters
+        ----------
+        cv : Union[str, BaseCrossValidator]
+            Cross validation parameter.
+        calibrator : Union[str, RegressorMixin]
+            Calibrator as string to then be linked to one of the
+            valid methods otherwise calibrator as estimator.
+
+        Returns
+        -------
+        RegressorMixin
+            RegressorMixin to be used as calibrator.
+        Raises
+        ------
+        ValueError
+            If str is not one of the valid estimators.
+        """
+        if calibrator is None:
+            calibrator = "sigmoid"
+
+        if isinstance(calibrator, str):
+            if calibrator in self.valid_calibrators.keys():
+                calibrator = self.valid_calibrators[calibrator]
+            else:
+                raise ValueError(
+                    "Please provide a valid string from the valid calibrators."
+                )
+        check_estimator_fit_predict(calibrator)
+        return calibrator
+
+    def _get_labels(
+        self,
+        X: ArrayLike
+    ) -> Tuple[NDArray, NDArray]:
+        """
+        The "labels" is the way to create the different arrays needed
+        for the type of calibration method you want to make.
+
+        - Top-Label method means that you take the maximum probability
+        and calibrated each class when it is the maximum separately.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray] of shapes (n_samples,) and (n_samples,)
+            The first element corresponds to the values that have to be
+            calibrated and the second element corresponds to class to be
+            associated.
+
+            In the Top-Label setting, the latter refers to the different
+            classes that have to be individually calibrated.
+
+        Raises
+        ------
+        ValueError
+            If the method provided has not been implemented.
+        """
+        pred = self.estimator.predict_proba(X=X)  # type: ignore
+        max_class_prob = np.max(pred, axis=1).reshape(-1, 1)
+        y_pred = self.classes_[np.argmax(pred, axis=1)]
+        return max_class_prob, y_pred
+
+    def _check_method(self) -> None:
+        """
+        This method checks that the method is valid.
+
+        Raises
+        ------
+        ValueError
+            If the method is not part of the valid methods.
+        """
+        if self.method not in self.valid_methods:
+            raise ValueError(
+                "Invalid method, allowed method are ", self.valid_methods
+            )
+
+    def _fit_calibrators(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike],
+        calibrator: RegressorMixin,
+    ) -> Dict[int, RegressorMixin]:
+        """
+        This method gets the correct labels and then fits in a loop
```

Comment:
Unclear, I think you mean that the method sequentially fit the calibrators for each class?

Reply:
done.